bpf/Makefile

```diff
@@ -189,6 +189,7 @@ testdata:
 
 generate: $(BPF)
 	$(GO) generate ../pkg/datapath/config
+	BPF2GO_CC="$(CLANG)" BPF2GO_CFLAGS="$(MAX_OVERLAY_OPTIONS) $(CLANG_FLAGS)" $(GO) generate ../pkg/datapath/loader
```

Comment:
This needs to be executed within the context of our LLVM toolchain image, otherwise this will be executed using the user's native toolchain, which is different from what CI will use while checking for drift.

If it's more convenient, the whole `generate` target can be run within the toolchain image. This was previously omitted for speed, but if we spin up a container anyway, this can be lobbed in together.

Reply:
Scratch that, I'm creating https://github.com/cilium/image-tools/pull/364 to add `llvm-strip` to the builder image to unblock this PR. Then yeah, we can run the generate steps (including `bpf2go`) inside that image.